### ISSUE TITLE
RTD build is broken

### ISSUE THREAD

**Issue by waprin** [GoogleCloudPlatform/google-cloud-python#3025]
Can look at this, leaving as note as reminder.

**Comment by daspecster** [comment]
I'm not sure what this means?

They appear to be populated?
https://googlecloudplatform.github.io/google-cloud-python/latest/storage-client.html

**Comment by waprin** [comment]
Sorry opened issue with customer next to me. 

@daspecster realizing its' the read the docs docs that are missing.

https://google-cloud-python.readthedocs.io/en/latest/storage-client.html

**Comment by dhermes** [comment]
Indeed, another example: https://google-cloud-python.readthedocs.io/en/latest/datastore-transactions.html

Latest build is green (but must be secretly not green): https://readthedocs.org/projects/google-cloud-python/builds/5028739/

**Comment by dhermes** [comment]
@tseaver @daspecster We should tackle this. I'm not sure if there is another issue, but the RTD build has been b0rken for a few days:

https://readthedocs.org/projects/google-cloud-python/builds/

e.g.

https://readthedocs.org/projects/google-cloud-python/builds/5060236/

**Comment by dhermes** [comment]
OK the fix is easy, we just need to get all subpackages on the same version of `google-cloud-core`. DERP!

**Comment by bjwatson** [comment]
This is a P0 issue that must be fixed ASAP. It's a poor user experience.

How can we avoid this with future `google-cloud-core` releases?

**Comment by daspecster** [comment]
@bjwatson #3056 should do it AFAICT.

**Comment by dhermes** [comment]
@bjwatson It's not the release, it's just a broken build.

**Comment by daspecster** [comment]
Looks like it's still broken :(
https://readthedocs.org/projects/google-cloud-python/builds/5066476/

**Comment by dhermes** [comment]
Jinx.

I think it's because of cached deps, so i'm blowing up the cache.

**Comment by dhermes** [comment]
It **was** because of cached deps, but now it's because of an accidental [missing comma][1] in [`proto-google-cloud-pubsub-v1==0.15.0`][2].

@lukesneeringer is **on the case**!

[1]: https://github.com/googleapis/api-client-staging/blob/d0682de146e60af7fe1084d8b1a9906c63e5050b/generated/python/proto-google-cloud-pubsub-v1/setup.py#L13
[2]: https://pypi.python.org/pypi/proto-google-cloud-pubsub-v1/0.15.0

**Comment by lukesneeringer** [comment]
Waiting on an LGTM and I will push the packages.

**Comment by lukesneeringer** [comment]
Packages to be added / pushed to 0.15.1 (proto only):

  * [ ] datastore (0.90.1)
  * [ ] error reporting
  * [ ] functions
  * [ ] language
  * [ ] monitoring
  * [ ] pubsub
  * [ ] spanner-admin-database
  * [ ] spanner-admin-instance
  * [ ] spanner
  * [ ] speech
  * [ ] vision

**Comment by dhermes** [comment]
Whoa that's a lot! Thanks @lukesneeringer

**Comment by lukesneeringer** [comment]
Sooner or later I am going to automate this... :-)

**Comment by lukesneeringer** [comment]
All launched except Speech.

**Comment by lukesneeringer** [comment]
Speech launched.

**Comment by dhermes** [comment]
Builds still failing, but it's because the umbrella package refers to versions other than the ones in source (this is an argument against getting too far out of sync)

**Comment by lukesneeringer** [comment]
Okay. Let's make a concerted effort to get this across the finish line tomorrow morning.

**Comment by dhermes** [comment]
Still b0rken but now it's not our fault: https://readthedocs.org/projects/google-cloud-python/builds/5071900/

```
pkg_resources.UnknownExtra: proto-google-cloud-datastore-v1 0.90.1 has no such extra feature 'grpc'
```

@lukesneeringer and I are aware of this, but we weren't aware it'd keep RTD from building

**Comment by bjwatson** [comment]
At least the `/latest/` RTD is showing content now: https://googlecloudplatform.github.io/google-cloud-python/latest/datastore-usage.html

**Comment by daspecster** [comment]
@bjwatson I was looking at `/latest/` the other day and it was fine? Was there a time when it wasn't showing content?

**Comment by dhermes** [comment]
@bjwatson That isn't RTD, this is: http://google-cloud-python.readthedocs.io/en/latest/datastore-usage.html

**Comment by bjwatson** [comment]
Ugh, you're right. Sorry, I'm juggling too many things at once right now.

Is there a plan to resolve the remaining issue?

**Comment by bjwatson** [comment]
@daspecster Yes.

**Comment by lukesneeringer** [comment]
Yeah, I know what needs to happen to fix that.

**Comment by bjwatson** [comment]
Ok, cool.

**Comment by lukesneeringer** [comment]
Correction: It is every `proto-` package, so after I fix it I will have to regenerate all of them, and every GAPIC.

**Comment by bjwatson** [comment]
Got it. Will we make `grpc` an option for every `proto-` package?

**Comment by lukesneeringer** [comment]
Yeah, that is exactly what needs to happen. We thought we were thorough when we changed grpc to optional but clearly I made a small mistake here.

**Comment by lukesneeringer** [comment]
Another note: A high priority needs to be that RTD builds from CI. We need to know about failures without having to check RTD.

**Comment by lukesneeringer** [comment]
Correction to my correction: Just Datastore.

**Comment by dhermes** [comment]
@lukesneeringer We do build `docs` on CI, we just use different flags than RTD uses.

**Comment by daspecster** [comment]
We had one build pass a couple days ago but we're back to this it looks like.
```
pkg_resources.UnknownExtra: proto-google-cloud-datastore-v1 0.90.1 has no such extra feature 'grpc'
```
See: https://readthedocs.org/projects/google-cloud-python/builds/5080674/

**Comment by lukesneeringer** [comment]
@dhermes You said this is fixed, correct?

**Comment by dhermes** [comment]
As of https://readthedocs.org/projects/google-cloud-python/builds/5081679/ it seems to be fixed (a new build after it also passed: https://readthedocs.org/projects/google-cloud-python/builds/5081815/)

**Comment by chlela** [comment]
Is it fair to say that documentation links are broken again?

https://googlecloudplatform.github.io/google-cloud-python/latest/pubsub/usage.html throws a 404 error

**Comment by dhermes** [comment]
@chlela No, that URL no longer has documentation (Pub/Sub recently had a very large change, docs included).
